python/dgl/distgnn/partition/libra_partition.py

```diff
@@ -0,0 +1,577 @@
+    nc = args.num_parts
+    prefix = args.out_dir
+    
+    print("dataset: ", dataset)
+    print("num partitions: ", nc)
+    print("output location: ", prefix)
+
+    index = 0
+    if dataset == 'cora':
+        resultdir = os.path.join(prefix, 'Libra_result_cora')
+    elif dataset == 'pubmed':
+        resultdir = os.path.join(prefix, 'Libra_result_pubmed')
+    elif dataset == 'citeseer':
+        resultdir = os.path.join(prefix, 'Libra_result_citeseer')
+    elif dataset == 'reddit':
+        resultdir = os.path.join(prefix, 'Libra_result_reddit')
+    elif dataset == 'ogbn-products':
+        resultdir = os.path.join(prefix, 'Libra_result_ogbn-products')
+        index = 1
+    elif dataset == 'ogbn-papers100M':
+        resultdir = os.path.join(prefix, 'Libra_result_ogbn-papers100M')
+        index = 3
+    elif dataset == 'proteins':
+        resultdir = os.path.join(prefix, 'Libra_result_proteins')
+        index = 2
+    elif dataset == 'ogbn-arxiv':
+        resultdir = os.path.join(prefix, 'Libra_result_ogbn-arxiv')
+    else:
+        raise DGLError("Error: Input dataset {}  not found !!", dataset)
+        
+    ## create ouptut directory
+    try:
+        os.makedirs(resultdir, mode=0o775, exist_ok=True)
+    except:
+        raise DGLError("Error: Could not create directory: ", resultdir)
+
+    ## Partitions per dataset 
+    l = [[2,4,8,16], [2,4,8,16,32,64],[2,4,8,16,32,64],[32,64,128]]
+    print("Output is stored in ", resultdir, flush=True)
+    #print("Generating ", l[index], " partitions...", flush=True)
+    print("Generating ", nc, " partitions...", flush=True)
+
+    tic = time.time()
+    #for num_community in l[index]:
+    for i in range(1):
+        num_community = nc    ## num communities or num partitions
+        print("####################################################################")
+        print("Executing parititons: ", num_community)
+        ltic = time.time()
+        try:
+            resultdir_libra2dgl = os.path.join(resultdir, str(num_community) + "Communities")
+            os.makedirs(resultdir_libra2dgl, mode=0o775, exist_ok=True)
+        except:
+            raise DGLError("Error: Could not create sub-directory: ", resultdir_libra2dgl)
+
+        ## Libra partitioning
+        max_weightsum  = vertex_cut_partition(num_community, dataset, resultdir)
+
+        print(" ** Converting libra partitions to dgl graphs **")
+        libra2dgl.run(dataset, resultdir_libra2dgl, num_community)
+        print("Conversion libra2dgl completed !!!")
+        ltoc = time.time()
+        print("Time taken by {} partitions {:0.4f} sec".format(num_community, ltoc - ltic))
+        print()
+
+    toc = time.time()
+    #print("Generated ", l[index], " partitions in {:0.4f} sec".format(toc - tic), flush=True)
+    print("Generated ", nc, " partitions in {:0.4f} sec".format(toc - tic), flush=True)
+    print("Partitioning completed successfully !!!")
+
+
+
+def partition_graph(dataset, nc, prefix):
```

Comment:
Right now both parts (structure partitioning and creating the ndata dictionaries) of this script use built-in DGL functions to load the datasets. Given a generic load function for dataset loading, it will be easier to write the data loading part of the script. Does DGL provide any such function?